### PR TITLE
Update exporter OT -> AI mappings based off of common schema

### PR DESCRIPTION
**Pull request by lzchen**
Based off of: https://github.com/microsoft/common-schema/blob/main/Mappings/AzureMonitor-AI.md

Used [OT semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/) and [common schema OT semantic conventions](https://github.com/microsoft/common-schema/blob/main/Mappings/OTelSemanticConvention.md) as references.

### CODE REVIEW

**Review by rakshith91 — APPROVED** [review]

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py [hunk 1]
```diff
@@ -101,41 +101,49 @@ def _convert_span_to_envelope(span: Span) -> TelemetryItem:
...
     if span.kind in (SpanKind.CONSUMER, SpanKind.SERVER):
         envelope.name = "Microsoft.ApplicationInsights.Request"
         data = RequestData(
-            name=span.name,
+            name=span.name[:1024],  # Breeze max length
             id="{:016x}".format(span.context.span_id),
             duration=_utils.ns_to_duration(span.end_time - span.start_time),
-            response_code=str(span.status.status_code.value),
+            response_code="0",
             success=span.status.is_ok,
```
Comment (hectorhdzg) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py:122]: Unset is also considered successful
https://github.com/microsoft/common-schema/blob/main/PartB/traces/Span.md#success

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py [hunk 2]
```diff
@@ -162,76 +176,142 @@ def _convert_span_to_envelope(span: Span) -> TelemetryItem:
...
+def _get_default_port_http(scheme):
+    if scheme == "http":
+        return 80
+    if scheme == "https":
+        return 443
+    return 0
+
+
+def _is_relational_db(dbsystem):
+    return dbsystem in ["postgresql", "mysql"]
```
Comment (hectorhdzg) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py:317]: There are way more relational DBs available in this property https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#connection-level-attributes do we want to add this "SQL" type, is kidna weird that we do not use the dbSystem always if available

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py [hunk 3]
```diff
@@ -162,76 +176,142 @@ def _convert_span_to_envelope(span: Span) -> TelemetryItem:
...
-                    data.target = "{}:{}".format(
-                        name,
-                        str(span.attributes["net.peer.port"]),
-                    )
-                elif "http.url" in span.attributes:
+                scheme = span.attributes.get("http.scheme")
+                if "http.url" in span.attributes:
                     url = span.attributes["http.url"]
                     # data is the url
                     data.data = url
```
Comment (hectorhdzg) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py:201]: There is some fallback logic when "http.url" is not available for data.data(httpUrl) https://github.com/microsoft/common-schema/blob/main/Mappings/OTelSemanticConvention.md#mapping-for-httpurl

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py (hunk 1 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py:122]: `is_ok` returns true if status is `unset`. https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/src/opentelemetry/trace/status.py#L77

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py (hunk 3 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py]: Added.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py (hunk 2 above)
Comment (hectorhdzg) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py:380]: Ok, I will follow up on spec discussions, having this kind of check seems unnecessary unless Breeze do something specific when type = "SQL"

**Review by hectorhdzg — APPROVED** [review]

**Pull request merged**